datafusion/expr-common/src/signature.rs

```diff
@@ -865,6 +867,39 @@ impl Signature {
             volatility,
         }
     }
+
+    /// Specialized Signature for ArrayPrepend and similar functions
+    pub fn element_and_array(volatility: Volatility) -> Self {
+        Signature {
+            type_signature: TypeSignature::ArraySignature(
+                ArrayFunctionSignature::Array {
+                    arguments: vec![
+                        ArrayFunctionArgument::Element,
+                        ArrayFunctionArgument::Array,
+                    ],
+                    array_coercion: Some(ListCoercion::FixedSizedListToList),
```

Comment:
tbh I'm not entirely clear on how the coercion comes into play but an append to a fixed size list would still be a fixed size list right?

Reply:
Generally if the list size is changed like (append operation) we will convert it to List. so append with fixed-size-list ends up with List

Reply:
What we need is to provide customizable signature, so if they want List for append, they can add `array_coercion: Some(ListCoercion::FixedSizedListToList)`, if they want to keep fixed-size-list they can add `array_coercion: None`. Both cases should be possible.

Reply:
Yep, that's why I'm setting `array_coercion: Some(ListCoercion::FixedSizedListToList)` by default 👍